ignite/chainconfig/parser.go

```diff
@@ -0,0 +1,239 @@
+package chainconfig
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/ignite-hq/cli/ignite/chainconfig/common"
+	"github.com/ignite-hq/cli/ignite/chainconfig/conversion"
+	v0 "github.com/ignite-hq/cli/ignite/chainconfig/v0"
+	v1 "github.com/ignite-hq/cli/ignite/chainconfig/v1"
+	"github.com/ignite-hq/cli/ignite/pkg/xfilepath"
+	"github.com/imdario/mergo"
+)
+
+var (
+	// ErrCouldntLocateConfig returned when config.yml cannot be found in the source code.
+	ErrCouldntLocateConfig = errors.New(
+		"could not locate a config.yml in your chain. please follow the link for" +
+			"how-to: https://github.com/ignite-hq/cli/blob/develop/docs/configure/index.md")
+
+	// ConfigDirPath returns the path of configuration directory of Ignite.
+	ConfigDirPath = xfilepath.JoinFromHome(xfilepath.Path(".ignite"))
+
+	// ConfigFileNames is a list of recognized names as for Ignite's config file.
+	ConfigFileNames = []string{"config.yml", "config.yaml"}
+
+	// Migration defines the version as the key and the config instance as the value
+	Migration = map[common.Version]common.Config{0: &v0.Config{}, 1: &v1.Config{}}
```

Comment:
Can we move this to a new file called chainconfig/config.go?

Can we also put each item in this map to a newline?

Reply:
done